core/cursor_svg.js

```diff
@@ -237,6 +243,37 @@ Blockly.CursorSvg.prototype.showWithField_ = function() {
   this.showCurrent_();
 };
 
+/**
+ * Show the cursor using a stack.
+ * @private
+ */
+Blockly.CursorSvg.prototype.showWithStack_ = function() {
+  var block = this.getCurNode().getLocation();
+
+  //Gets the height and width of entire stack
+  var heightWidth = block.getHeightWidth();
+
+  //Add padding so that being on a stack looks different than being on a block
+  var width = heightWidth.width + Blockly.CursorSvg.STACK_PADDING;
+  var height = heightWidth.height + Blockly.CursorSvg.STACK_PADDING;
+
+  //Shift the rectangle slightly to upper left so padding is equal on all sides
+  var x =  -1 * Blockly.CursorSvg.STACK_PADDING / 2;
+  var y =  -1 * Blockly.CursorSvg.STACK_PADDING / 2;
+
+  //If the block has an output connection it needs more padding
+  if (block.outputConnection) {
+    x = -1 * Blockly.CursorSvg.STACK_PADDING;
```

Comment:
Should this be x -= Blockly.BlockSvg.TAB_WIDTH; and do you need to add the tab width to the cursor width or is it calculated in the block's width but drawn with a - offset?

Reply:
Yes to the first part, but I think the TAB_WIDTH is included when we get the width for the entire block so we just need to change where it is positioned.